pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/ChildAggregationFunction.java

```diff
@@ -0,0 +1,148 @@
+    expressionContexts.add(_childFunctionID);
+    expressionContexts.add(_childFunctionKeyInParent);
+    expressionContexts.addAll(_resultNameOperands);
+    return expressionContexts;
+  }
+
+  @Override
+  public final AggregationResultHolder createAggregationResultHolder() {
+    return new DummyAggregationResultHolder();
+  }
+
+  @Override
+  public final GroupByResultHolder createGroupByResultHolder(int initialCapacity, int maxCapacity) {
+    return new DummyGroupByResultHolder();
+  }
+
+  @Override
+  public final void aggregate(int length, AggregationResultHolder aggregationResultHolder,
+      Map<ExpressionContext, BlockValSet> blockValSetMap) {
+  }
+
+  @Override
+  public final void aggregateGroupBySV(int length, int[] groupKeyArray, GroupByResultHolder groupByResultHolder,
+      Map<ExpressionContext, BlockValSet> blockValSetMap) {
+  }
+
+  @Override
+  public final void aggregateGroupByMV(int length, int[][] groupKeysArray, GroupByResultHolder groupByResultHolder,
+      Map<ExpressionContext, BlockValSet> blockValSetMap) {
+  }
+
+  @Override
+  public final Long extractAggregationResult(AggregationResultHolder aggregationResultHolder) {
+    return 0L;
+  }
+
+  @Override
+  public final Long extractGroupByResult(GroupByResultHolder groupByResultHolder, int groupKey) {
+    return 0L;
+  }
+
+  @Override
+  public final Long merge(Long intermediateResult1, Long intermediateResult2) {
+    return 0L;
+  }
+
+  @Override
+  public final DataSchema.ColumnDataType getIntermediateResultColumnType() {
+    return DataSchema.ColumnDataType.LONG;
+  }
+
+  @Override
+  public final DataSchema.ColumnDataType getFinalResultColumnType() {
+    return DataSchema.ColumnDataType.UNKNOWN;
+  }
+
+  @Override
+  public final Long extractFinalResult(Long longValue) {
+    return 0L;
+  }
+
+  /**
+   * The name of the column as follows:
+   * CHILD_AGGREGATION_NAME_PREFIX + actual function type + operands + CHILD_AGGREGATION_SEPERATOR
+   * + actual function type + parent aggregation function id + CHILD_KEY_SEPERATOR + column key in parent function
+   */
+  @Override
+  public final String getResultColumnName() {
+    String type = getType().getName().toLowerCase();
+    return CommonConstants.RewriterConstants.CHILD_AGGREGATION_NAME_PREFIX
+        // above is the prefix for all child aggregation functions
+
+        + type + "(" + _resultNameOperands.stream().map(ExpressionContext::toString)
+        .collect(Collectors.joining(",")) + ")"
+        // above is the actual child aggregation function name we want to return to the user
+
+        + CommonConstants.RewriterConstants.CHILD_AGGREGATION_SEPERATOR
+        + type
+        + _childFunctionID.getLiteral().getStringValue()
+        + CommonConstants.RewriterConstants.CHILD_KEY_SEPERATOR
+        + _childFunctionKeyInParent.toString();
+    // above is the column key in the parent aggregation function
+  }
+
+  @Override
+  public String toExplainString() {
```

Comment:
Recommend adding a test or two to the `ExplainPlanQueriesTest` file to test out the EXPLAIN PLAN output

Reply:
added